fs/operations/operations.go

```diff
@@ -336,6 +336,18 @@ func Copy(ctx context.Context, f fs.Fs, dst fs.Object, remote string, src fs.Obj
 	doUpdate := dst != nil
 	hashType, hashOption := CommonHash(ctx, f, src.Fs())
 
+	remotePartial := remote
+	// TODO don't set remotePartial if --inplace=true
+	if f.Features().Move != nil && f.Features().PartialUploads {
+		if len(path.Base(remote)) < 100 {
+			remotePartial = remote + ".partial-" + random.String(8)
+		} else {
+			// Don't make the leaf name longer if it's already lengthy to avoid
+			// trouble with file name length limits.
+			fs.Debugf(src, "Not using partial download/rename due to long file name length")
+		}
```

Comment:
I've been in a kind of coffee induced brain fog.  Your suggestion is better.  I was somehow stuck on not allowing the original file name to be truncated, but that doesn't matter in this case.

Reply:
I changed it a little as the above was generating a negative index.  So I'm truncating the start of the string if it's too long:

```		
remotePartial = remote + ".partial-" + random.String(8)
if len(remotePartial) > 100 {
	remotePartial = remotePartial[len(remotePartial)-100:]
}
```

Reply:
Remember the remote string is the whole file path (eg `dir/subdir/file.txt`)  so may have directories in. In this case truncating the start isn't a good idea!

Reply:
Uhh, yeah, it absolutely isn't! I think this logic should be made to work with basenames instead of the full path.

Reply:
Like this maybe:

```
	remotePartial := remote
	if !ci.Inplace && f.Features().Move != nil && f.Features().PartialUploads {
		// Avoid making the leaf name longer if it's already lengthy to avoid
		// trouble with file name length limits.
		suffix := ".partial-" + random.String(8)
		dir, base := path.Split(remote)
		baseSuffix := base + suffix
		if len(baseSuffix) > 100 {
			remotePartial = path.Join(dir, baseSuffix[len(baseSuffix)-100:])
		} else {
			remotePartial = remote + suffix
		}
	}
```